osf/models/user.py

```diff
@@ -1384,22 +1394,23 @@ def add_unclaimed_record(self, claim_origin, referrer, given_name, email=None):
         """
 
         from osf.models.provider import AbstractProvider
-        from osf.models import Preprint
+        from osf.models.osf_group import OSFGroup
+
         if isinstance(claim_origin, AbstractProvider):
             if not bool(get_perms(referrer, claim_origin)):
                 raise PermissionsError(
                     'Referrer does not have permission to add a moderator to provider {0}'.format(claim_origin._id)
                 )
 
-        elif isinstance(claim_origin, Preprint):
-            if not claim_origin.has_permission(referrer, 'admin'):
+        elif isinstance(claim_origin, OSFGroup):
+            if not claim_origin.has_permission(referrer, 'manage'):
```

Comment:
When adding an unregistered group member, the OSF group id is added to `unclaimed_records`.